src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -253,7 +253,7 @@ export class CalciteRadioButton {
     this.input.checked = this.checked;
     this.input.disabled = this.disabled;
     this.input.hidden = this.hidden;
-    this.input.id = this.guid;
+    this.input.id = `${this.guid}-input`;
```

Comment:
The `id` of the hidden input shouldn't be the same as the `id` on the host, so I updated this as well as the `id` on checkbox to support the `for` behavior for `calcite-label`.

Reply:
re: https://github.com/Esri/calcite-components/issues/871 - do we need to set guids at all - can folks get the element directly if needed?

Reply:
The `guid` is particularly important for `calcite-radio-button` because it relies on it to uncheck other radios in the same group when it is clicked: https://github.com/Esri/calcite-components/blob/825/calcite-label-form-accessibility/src/components/calcite-radio-button/calcite-radio-button.tsx#L162

The `guid` is also just a fallback when an `id` is not supplied, so if a user wants to access the element directly, all they would need to do is set an `id` on the `calcite-radio-button` and use `document.querySelector(`${id}-input`)` to gain access to that.